core/Controller/TaskProcessingApiController.php

```diff
@@ -493,4 +529,15 @@ public function getNextScheduledTask(array $providerIds, array $taskTypeIds): Da
 			return new DataResponse(['message' => $this->l->t('Internal error')], Http::STATUS_INTERNAL_SERVER_ERROR);
 		}
 	}
+
+	private function setFileContentsInternal(Task $task, string $data) {
+		try {
+			$folder = $this->appData->getFolder('TaskProcessing');
+		} catch (\OCP\Files\NotFoundException) {
+			$folder = $this->appData->newFolder('TaskProcessing');
+		}
+		/** @var SimpleFile $file */
+		$file = $folder->newFile((string) rand(0, 10000000), $data);
+		return $file->getId();
```

Comment:
What happens if the file already exists? Will the file be overwritten or will it fail?

Reply:
from the provided code it is obvious: new file will be created

Reply:
```suggestion
		/** @var SimpleFile $file */
		$file = $folder->newFile(time() . '-' . rand(0, 10000000), $data);
		return $file->getId();
```

May be better to use current time for the file name as well.